Backoffice/Form/Type/Component/RoleChoiceType.php

```diff
@@ -13,15 +15,32 @@ class RoleChoiceType extends AbstractType
 {
     protected $roleCollector;
     protected $name;
+    protected $parameters;
 
     /**
      * @param RoleCollectorInterface $roleCollector
      * @param string                 $name
      */
-    public function __construct(RoleCollectorInterface $roleCollector, $name)
+    public function __construct(RoleCollectorInterface $roleCollector, $name, $parameters)
     {
         $this->roleCollector = $roleCollector;
         $this->name = $name;
+        $this->parameters = $parameters;
+    }
+
+    /**
+     * @param FormView      $view
+     * @param FormInterface $form
+     * @param array         $options
+     */
+    public function buildView(FormView $view, FormInterface $form, array $options)
+    {
+        $parameters = array();
+        $choices = $this->roleCollector->getRoles();
+        foreach($this->parameters as $key => $parameter) {
+            $parameters[$parameter['category']][$parameter['label']][] = array_search(strtoupper($key), array_keys($choices));
```

Comment:
There is two many $parameter(s) in the class, it makes the code not easy to understand. Can you find more explicit names ? Which kind of parameter(s) these are ?